<a href="https://colab.research.google.com/github/DirecTV-Team1A-NextGen-Rec/directv_tvrec/blob/main/DirecTV_Data_Understanding_Serena.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Serena Kim (9.11.2024)

In [28]:
import pandas as pd
import os
from IPython.display import display
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

Upload the TMDB_tv_dataset_v3.csv file.

In [42]:
# from google.colab import files
# uploaded = files.upload()

### 1. Build DataFrame

In [30]:
TMDB_filename = os.path.join(os.getcwd(), "TMDB_tv_dataset_v3.csv")
df = pd.read_csv(TMDB_filename)

 ### 2. Basic Exploration of TMDB Data

In [31]:
#dataframe shape
df.shape

(168639, 29)

In [32]:
#first few rows:

display(df.head())

id              name  number_of_seasons  number_of_episodes  \
0   1399   Game of Thrones                  8                  73   
1  71446       Money Heist                  3                  41   
2  66732   Stranger Things                  4                  34   
3   1402  The Walking Dead                 11                 177   
4  63174           Lucifer                  6                  93   

  original_language  vote_count  vote_average  \
0                en       21857         8.442   
1                es       17836         8.257   
2                en       16161         8.624   
3                en       15432         8.121   
4                en       13870         8.486   

                                            overview  adult  \
0  Seven noble families fight for control of the ...  False   
1  To carry out the biggest heist in history, a m...  False   
2  When a young boy vanishes, a small town uncove...  False   
3  Sheriff's deputy Rick Grimes awakens from a co...  False   
4  Bored and unhappy as the Lord of Hell, Lucifer...  False   

                      backdrop_path  ...                           tagline  \
0   /2OMB0ynKlyIenMJWI2Dy9IWT4c.jpg  ...                  Winter Is Coming   
1  /gFZriCkpJYsApPZEF3jhxL4yLzG.jpg  ...              The perfect robbery.   
2  /2MaumbgBlW1NoPo3ZJO38A6v7OS.jpg  ...     Every ending has a beginning.   
3  /x4salpjB11umlUOltfNvSSrjSXm.jpg  ...  Fight the dead. Fear the living.   
4  /aDBRtunw49UF4XmqfyNuD9nlYIu.jpg  ...              It's good to be bad.   

                                        genres                 created_by  \
0  Sci-Fi & Fantasy, Drama, Action & Adventure  David Benioff, D.B. Weiss   
1                                 Crime, Drama                  Álex Pina   
2             Drama, Sci-Fi & Fantasy, Mystery   Matt Duffer, Ross Duffer   
3  Action & Adventure, Drama, Sci-Fi & Fantasy             Frank Darabont   
4                      Crime, Sci-Fi & Fantasy                Tom Kapinos   

   languages           networks  origin_country spoken_languages  \
0         en                HBO              US          English   
1         es  Netflix, Antena 3              ES          Español   
2         en            Netflix              US          English   
3         en                AMC              US          English   
4         en       FOX, Netflix              US          English   

                                production_companies  \
0  Revolution Sun Studios, Television 360, Genera...   
1                                    Vancouver Media   
2  21 Laps Entertainment, Monkey Massacre Product...   
3  AMC Studios, Circle of Confusion, Valhalla Mot...   
4  Warner Bros. Television, DC Entertainment, Jer...   

                       production_countries episode_run_time  
0  United Kingdom, United States of America                0  
1                                     Spain               70  
2                  United States of America                0  
3                  United States of America               42  
4                  United States of America               45  

[5 rows x 29 columns]

In [33]:
#last few rows:
display(df.tail())

id                 name  number_of_seasons  number_of_episodes  \
168634  239099             母乳酱想要喷出来                  1                   1   
168635  241205   Barbie Dream Squad                  1                   1   
168636  240696              Picasso                  1                   1   
168637  240697         女子大生危険なアルバイト                  1                   1   
168638  237566  Welcome to My World                  1                   5   

       original_language  vote_count  vote_average  \
168634                zh           0           0.0   
168635                es           0           0.0   
168636                bn           0           0.0   
168637                en           0           0.0   
168638                en           0           0.0   

                                                 overview  adult  \
168634                                                NaN  False   
168635                                                NaN  False   
168636  Murder, art and a journalist's relentless ques...  False   
168637  A college student gets into trouble when she m...  False   
168638                 Short-lives series on Youtube Red.  False   

                          backdrop_path  ... tagline genres  \
168634                              NaN  ...     NaN    NaN   
168635                              NaN  ...     NaN    NaN   
168636  /fseBBbvHtdcZS5M7bSjOxYsVe5.jpg  ...     NaN  Crime   
168637                              NaN  ...     NaN    NaN   
168638                              NaN  ...     NaN    NaN   

                         created_by  languages networks  origin_country  \
168634                          NaN        NaN      NaN              CN   
168635  Dee Shipley, Blair Davidson        NaN      NaN              US   
168636                          NaN         bn      NaN              IN   
168637                          NaN        NaN      NaN              US   
168638              Paul Vandervort        NaN      NaN              US   

       spoken_languages production_companies      production_countries  \
168634              NaN                  NaN                       NaN   
168635              NaN                  NaN                       NaN   
168636            বাংলা                  NaN                     India   
168637              NaN                  NaN                       NaN   
168638              NaN          YouTube Red  United States of America   

       episode_run_time  
168634                0  
168635                0  
168636                0  
168637                0  
168638                0  

[5 rows x 29 columns]

In [34]:
# display basic statistics of numeric columns

display(df.describe())

id  number_of_seasons  number_of_episodes     vote_count  \
count  168639.000000      168639.000000       168639.000000  168639.000000   
mean   111307.074704           1.548497           24.465082      13.305054   
std     76451.662352           2.942872          134.799622     190.809059   
min         1.000000           0.000000            0.000000       0.000000   
25%     45936.500000           1.000000            1.000000       0.000000   
50%     97734.000000           1.000000            6.000000       0.000000   
75%    196923.500000           1.000000           20.000000       1.000000   
max    251213.000000         240.000000        20839.000000   21857.000000   

        vote_average     popularity  episode_run_time  
count  168639.000000  168639.000000     168639.000000  
mean        2.333843       5.882644         22.603348  
std         3.454334      42.023216         47.950427  
min         0.000000       0.000000          0.000000  
25%         0.000000       0.600000          0.000000  
50%         0.000000       0.857000          0.000000  
75%         6.000000       2.431500         42.000000  
max        10.000000    3707.008000       6032.000000

In [35]:
# display info about DataFrame
display(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 168639 entries, 0 to 168638
Data columns (total 29 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   id                    168639 non-null  int64  
 1   name                  168634 non-null  object 
 2   number_of_seasons     168639 non-null  int64  
 3   number_of_episodes    168639 non-null  int64  
 4   original_language     168639 non-null  object 
 5   vote_count            168639 non-null  int64  
 6   vote_average          168639 non-null  float64
 7   overview              93333 non-null   object 
 8   adult                 168639 non-null  bool   
 9   backdrop_path         77780 non-null   object 
 10  first_air_date        136903 non-null  object 
 11  last_air_date         138735 non-null  object 
 12  homepage              50998 non-null   object 
 13  in_production         168639 non-null  bool   
 14  original_name         168634 non-null  object 
 15  

None

In [36]:
#identifying features
df.columns.tolist()

['id',
 'name',
 'number_of_seasons',
 'number_of_episodes',
 'original_language',
 'vote_count',
 'vote_average',
 'overview',
 'adult',
 'backdrop_path',
 'first_air_date',
 'last_air_date',
 'homepage',
 'in_production',
 'original_name',
 'popularity',
 'poster_path',
 'type',
 'status',
 'tagline',
 'genres',
 'created_by',
 'languages',
 'networks',
 'origin_country',
 'spoken_languages',
 'production_companies',
 'production_countries',
 'episode_run_time']

### 3. Finding Basic Patterns or Commonalities in Features

In [37]:
# display the most occuring original languages
original_language_counts = df["original_language"].value_counts()

with pd.option_context("display.max_rows", None):
  display(original_language_counts)

# Get the count of unique original languages
num_unique_languages = len(original_language_counts)

# Display the number of different languages
print(f"Number of different original languages: {num_unique_languages}")


original_language
en    76304
zh    14422
ja    14048
ko     7820
de     7712
fr     7290
es     5602
pt     3551
ru     2963
nl     2923
ar     2638
da     2019
th     1835
cn     1824
tr     1748
it     1691
hi     1565
sv     1455
cs     1382
no     1210
pl      923
el      903
tl      861
he      553
ms      495
id      459
fi      453
sk      404
hu      404
ur      333
fa      244
bn      229
vi      205
ca      156
sr      152
hr      121
uk      118
ta      114
ro      113
my      112
bg       93
af       88
te       84
nb       80
sh       65
lt       65
is       55
ka       54
et       52
sl       50
lv       49
am       48
ml       38
mr       37
kk       31
cy       30
bs       29
xx       28
pa       26
ga       25
mt       20
sq       20
gu       20
kn       20
gl       18
or       14
ab       13
eu       12
az       11
ku       10
zu       10
mo        7
hy        7
si        6
ne        6
la        5
mk        5
km        5
as        5
gd        4
mi        4
be        3
jv        3
fy        3
hz        2
so        2
dv        2
lb        2
av        2
ps        1
lo        1
ht        1
ln        1
kv        1
eo        1
ho        1
ti        1
ug        1
rm        1
uz        1
za        1
sw        1
st        1
mn        1
se        1
aa        1
Name: count, dtype: int64

Number of different original languages: 106


Findings:
- Shows in English were the most common.
- There were shows in 106 languages total.

In [38]:
# display the shows with the highest vote average
sorted_shows_by_vote_average = df[["name", "vote_count", "vote_average"]].sort_values(by="vote_average", ascending=False)

display(sorted_shows_by_vote_average)

name  vote_count  vote_average
51932     The Gerald McBoing-Boing Show           1          10.0
54800             She's Gone with Water           1          10.0
38845                           Tarikat           1          10.0
38859   Télévision (histoires secrètes)           1          10.0
38864            Pilgrimage of Wealth 2           1          10.0
...                                 ...         ...           ...
94501                              妻不厌诈           0           0.0
94500                              少年天医           0           0.0
94499                           世间星辰不及你           0           0.0
94498                Pappa på förskolan           0           0.0
168638              Welcome to My World           0           0.0

[168639 rows x 3 columns]

Finding:
- Not so accurate, since shows with highest rating had only one vote.

Next Step:
We will display shows with both a high vote count and a high vote average.



In [41]:
high_vote_average_threshold = 7
high_vote_count_threshold = 100

filtered_shows = df[(df["vote_average"] >= high_vote_average_threshold) &
                    (df["vote_count"] >= high_vote_count_threshold)]

sorted_filtered_shows = filtered_shows[["name", "vote_average", "vote_count"]].sort_values(
    by=["vote_average", "vote_count"], ascending=False)

display(sorted_filtered_shows)

total_shows = len(df)
filtered_shows_count = len(filtered_shows)

# percentage of shows with over 100 votes
percentage = (filtered_shows_count / total_shows) * 100
print(f"{percentage:.2f}% of shows have over {high_vote_count_threshold} votes")

# percentage of shows with only 1 vote or 0 votes
shows_with_zero_or_one_vote = df[df["vote_count"].isin([0, 1])]
number_of_shows_with_zero_or_one_vote = len(shows_with_zero_or_one_vote)
percentage_of_shows_with_zero_or_one_vote = (number_of_shows_with_zero_or_one_vote / total_shows) * 100

print(f"{percentage_of_shows_with_zero_or_one_vote:.2f}% of shows have either 0 or 1 votes")

name  vote_average  vote_count
7                       Breaking Bad         8.890       12398
85                            Arcane         8.740        3341
60                         One Piece         8.725        4052
21                    Rick and Morty         8.700        8595
34    Demon Slayer: Kimetsu no Yaiba         8.700        5618
...                              ...           ...         ...
2928                     Meant to Be         7.000         104
2948             Mi Pequeña Traviesa         7.000         103
2994                Como Tu No Hay 2         7.000         101
2996               Man Seeking Woman         7.000         101
3021                            Lexx         7.000         100

[2633 rows x 3 columns]

1.56% of shows have over 100 votes
79.25% of shows have either 0 or 1 votes
